kolibri/core/public/api.py

```diff
@@ -324,3 +327,35 @@ def create(self, request):
 
     def update(self, request, pk=None):
         return self.check_queue(request, pk=pk)
+
+
+class FacilitySearchUsernameViewSet(ReadOnlyValuesViewset):
+    filter_backends = (DjangoFilterBackend, filters.SearchFilter)
+    filter_fields = ("facility",)
+    search_fields = ("^username",)
+
+    values = ("id", "username")
+
+    def list(self, request, *args, **kwargs):
+        facility_id = request.query_params.get("facility", None)
```

Comment:
Not blocking, but could maybe return a 412 here, if the facility_id is required but not given.

Reply:
sure, much better. Done!